neurosynth/analysis/reduce.py

```diff
@@ -155,3 +160,159 @@ def get_random_voxels(dataset, n_voxels):
+    """ Perform topic modeling using Latent Dirichlet Allocation with the
+    Java toolbox MALLET.
+    
+    Args:
+        abstracts:  A pandas DataFrame with two columns ('pmid' and 'abstract')
+                    containing article abstracts.
+        n_topics:   Number of topics to generate. Default=50.
+        n_words:    Number of top words to return for each topic. Default=31,
+                    based on Poldrack et al. (2012).
+        n_iters:    Number of iterations to run in training topic model.
+                    Default=1000.
+        alpha:      The Dirichlet prior on the per-document topic distributions.
+                    Default: 50 / n_topics, based on Poldrack et al. (2012).
+        beta:       The Dirichlet prior on the per-topic word distribution.
+                    Default: 0.001, based on Poldrack et al. (2012).
+    
+    Returns:
+        weights_df: A pandas DataFrame derived from the MALLET
+                    output-doc-topics output file. Contains the weight assigned
+                    to each article for each topic, which can be used to select
+                    articles for topic-based meta-analyses (accepted threshold
+                    from Poldrack article is 0.001). [n_topics]+1 columns:
+                    'pmid' is the first column and the following columns are
+                    the topic names. The names of the topics match the names
+                    in df (e.g., topic_000).
+        keys_df:    A pandas DataFrame derived from the MALLET
+                    output-topic-keys output file. Contains the top [n_words]
+                    words for each topic, which can act as a summary of the
+                    topic's content. Two columns: 'topic' and 'terms'. The
+                    names of the topics match the names in weights (e.g.,
+                    topic_000).
+    """
+    if abstracts.index.name != 'pmid':
+        abstracts.index = abstracts['pmid']
+
+    resdir = os.path.abspath(get_resource_path())
+    tempdir = os.path.join(resdir, 'topic_models')
+    absdir = os.path.join(tempdir, 'abstracts')
+    if not os.path.isdir(tempdir):
+        os.mkdir(tempdir)
+    
+    if alpha is None:
+        alpha = 50. / n_topics
+    
+    # Check for presence of abstract files and convert if necessary
+    if not os.path.isdir(absdir):
+        print('Abstracts folder not found. Creating abstract files...')
+        os.mkdir(absdir)
+        for pmid in abstracts.index.values:
+            abstract = abstracts.loc[pmid]['abstract']
+            with open(os.path.join(absdir, str(pmid)+'.txt'), 'w') as fo:
+                fo.write(abstract)
+        
+    # Run MALLET topic modeling
+    print('Generating topics...')
+    import_str = ('{mallet} import-dir '
+                  '--input {absdir} '
+                  '--output {outdir}/topic-input.mallet '
+                  '--keep-sequence '
+                  '--remove-stopwords').format(mallet=mallet_bin,
```

Comment:
`mallet_bin` isn't defined anywhere, and throws an error in the example notebook. This can be internally set given that it's known to be in neurosynth/resources, right?